.github/workflows/prune.yml

```diff
@@ -10,19 +10,74 @@ on:
+        run: |
+          image_names=$(gh api \
+            -H "Accept: application/vnd.github+json" \
+            -H "-GitHub-Api-Version: 2022-11-28" \
+            -H "Authorization: Token ${GH_TOKEN}" \
+            --paginate "/orgs/boozallen/packages?package_type=container" | \
+            jq -c '[.[] | select((.name | startswith("aissemble")) and (.name | endswith("-chart") | not)) | .name]')
+          
+          # save to output
+          echo "matrix_values=${image_names}" >> $GITHUB_OUTPUT
+  
+          # loop through each image name and find all release tags using regex pattern
+          for name in $(echo "$image_names" | jq -r '.[]'); do
+              release_versions=$(gh api \
+                -H "Accept: application/vnd.github+json" \
+                -H "-GitHub-Api-Version: 2022-11-28" \
+                -H "Authorization: Token ${GH_TOKEN}" \
+                --paginate "/orgs/boozallen/packages/container/${name}/versions" \
+                | jq -r '.[] | .metadata.container.tags[] | select(test("^\\d+\\.\\d+\\.\\d+$"))')
+          
+              # append all release tags to all_release_versions array
+              all_release_versions+=("${release_versions}")
+          done
+          
+          # replace all whitespaces with newline (required for sort) and then sort all release versions while removing any duplicates
+          all_release_versions_sorted=$(echo "${all_release_versions[@]}" | tr ' ' '\n'| sort -u )
+          
+          # replace the newlines with commas and then remove the trailing comma
+          all_release_versions=$(echo "${all_release_versions_sorted[@]}" | tr '\n' ',' | sed 's/,$//')
+          
+          # save to output
+          echo "release_values=${all_release_versions}" >> $GITHUB_OUTPUT
+          
+
+    # Identify the outputs of phase 1: fetch_image_info so they can be used in phase 2
+    outputs:
+      matrix-values: ${{ steps.fetch-image-info.outputs.matrix_values }}
+      release-values: ${{ steps.fetch-image-info.outputs.release_values }}
+
+  
+  # Phase 2: loop through output of phase 1 to prune images
+  prune_ghcr:
+    # Tells workflow to wait for the "fetch_image_info" phase to finish
+    needs: fetch_image_info
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        package-name: ${{ fromJson(needs.fetch_image_info.outputs.matrix-values) }}
+
     steps:
-      - name: Prune
-        uses: snok/container-retention-policy@v2
+      - name: Prune ghcr
+        uses: dataaxiom/ghcr-cleanup-action@v1
         with:
-          image-names: aissemble-*
-          cut-off: Two days ago UTC
-          account-type: org
-          org-name: boozallen
-          keep-at-least: 2
-          untagged-only: true
-          dry-run: false
+          log-level: info
+          package: ${{ matrix.package-name }}
           token: ${{ secrets.GHCR_IO_TOKEN }}
+          exclude-tags: ${{ needs.fetch_image_info.outputs.release-values }}
```

Comment:
I: `exclude-tags` is set to the value of the `release-values` output from phase 1, which is a string containing a comma separated list of release versions (e.g., `1.7.0,1.8.0`)

Reply:
Seeing how you had to implement the exclude-tags with an external regex it makes me think regex support should be directly supported on the exclude-tags setting in the GitHub cleanup action. I'll work on adding that function in the ghcr-cleanup-action.

Reply:
thanks @rohanmars!! we had trouble finding a clean up action that supported multi-arch images and regex so that would be a great feature to add.

Reply:
Okay great. Yes, I had a hard time finding a multi-arch action so I ended up just building one, haha. I'll add regex support soon. I think it might make sense also to add a "packages" mode on the action so it could support different packages (images) without having todo a matrix. I think to think about that more.

Reply:
that would be great, thanks again!